contracts/Comet.sol

```diff
@@ -764,47 +784,27 @@ contract Comet is CometMainInterface {
      * @dev Note: Safely handles non-standard ERC-20 tokens that do not return a value. See here: https://medium.com/coinmonks/missing-return-value-bug-at-least-130-tokens-affected-d67bf08521ca
      */
     function doTransferIn(address asset, address from, uint amount) internal returns (uint) {
-        uint256 preTransferBalance = IERC20NonStandard(asset).balanceOf(address(this));
-        IERC20NonStandard(asset).transferFrom(from, address(this), amount);
-        bool success;
-        assembly {
-            switch returndatasize()
-                case 0 {                       // This is a non-standard ERC-20
-                    success := not(0)          // set success to true
-                }
-                case 32 {                      // This is a compliant ERC-20
-                    returndatacopy(0, 0, 32)
-                    success := mload(0)        // Set `success = returndata` of override external call
-                }
-                default {                      // This is an excessively non-compliant ERC-20, revert.
-                    revert(0, 0)
-                }
+        _nonReentrantBefore();
```

Comment:
Sorry, I meant inline the `_nonReentrant()` logic inside the `nonReentrant` modifier and then keep the modifier on these functions.

Reply:
Oh I see 😅  Updated it.